done

nataliSM opened the pull request.

nataliSM committed:
done

ZalyalovIldar commented on SwiftLesson2/AppDelegate.swift:
```diff
@@ -15,7 +15,18 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
+            loginViewController?.modalTransitionStyle = UIModalTransitionStyle.flipHorizontal
+            
+            window?.makeKeyAndVisible()
+            window?.rootViewController?.present(loginViewController!, animated: false, completion: nil)
```
guard-ом сделай unwrapping window-а, чтобы лешить опциональности, тип вдруг что)

ZalyalovIldar commented on SwiftLesson2/LoginManager.swift:
```diff
@@ -0,0 +1,61 @@
+        
+       let isRegistred = isValitedLoginData(login: login, password: password)
+        if isRegistred {
+            UserDefaults.standard.set(isRegistred, forKey: loginStatusKey)
```
synhronize не сделала 🐛

ZalyalovIldar commented on SwiftLesson2/LoginManager.swift:
```diff
@@ -0,0 +1,61 @@
+    }
+    
+    func logOut() {
+        UserDefaults.standard.set(false, forKey: loginStatusKey)
```
и тут synhronize не сделала 🐛

ZalyalovIldar commented on SwiftLesson2/LoginManager.swift:
```diff
@@ -0,0 +1,61 @@
+    }
+    
+    func loginStatus() -> Bool {
+        guard let status:Bool = UserDefaults.standard.value(forKey: loginStatusKey) as! Bool? else{
```
тут лучше было бы использовать if let думаю, потому что тебе нужно именно использовать и true и false версию, при этом по сути status переменная не нужна как и не нужно делать опциональным Bool?, и можно заменить например на такое: 

`if let _ =  UserDefaults.standard.value(forKey: loginStatusKey) as! Bool{
     return true 
}else{
    return false
}`

ZalyalovIldar commented on SwiftLesson2/LoginViewController.swift:
```diff
@@ -0,0 +1,40 @@
+        
+        
+        
+        let status:Bool = LoginManager.sharedInstance.login(login: loginTextField.text!,
```
LoginManager ненужно делать singleTon-ом, потому что тебе не нужно хранить состояние, достаточно вызывать методы класса

ZalyalovIldar commented on SwiftLesson2/LoginViewController.swift:
```diff
@@ -0,0 +1,40 @@
+        let status:Bool = LoginManager.sharedInstance.login(login: loginTextField.text!,
+                                                            password: passwordTextField.text!)
+        if status {
+            dismiss(animated: true, completion: nil)
```
не забывай, пожалуйста, использовать " self ", просто привыкай :)

ZalyalovIldar commented on SwiftLesson2/ViewController.swift:
```diff
@@ -8,68 +8,137 @@
-        self.latitute = latititute
-        self.longitute = longitute
-        self.name = "name"
+    init(imageName: String, text: String) {
```
init конечно смысла не было делать, потому что точно такой-же инит генерится по дефолту для твоей структуры, новый инит есть смысл делать, только если он будет какой-то кастомный, так что это лишний код тут :)

ZalyalovIldar commented on SwiftLesson2/ViewController.swift:
```diff
@@ -8,68 +8,137 @@
+
+    
+    @IBAction func logoutButonDidClicked(_ sender: UIBarButtonItem) {
+        let loginController =  storyboard?.instantiateViewController(withIdentifier: "loginController")
```
тут лучше использовать guard :)

ZalyalovIldar reviewed (changes requested):
В целом все отлично, можно было бы конечно заюзать дополнительные Extensions, но пока ладно) 

Жду правок + по всему коду поправь где не поставила " self ", да без этого работает, да минусов нет, но по код стайлу все таки нужно ставить, чтобы обозначать, что это методы класса, а не функции :)

И да, у тебя в проекте нет никакой структуры, задай пожалуйста :)

nataliSM committed:
Исправила недочеты 🤗

1) по поводу self:  читала code-style .
https://github.com/raywenderlich/swift-style-guide#use-of-self
Он пишет, что self нужно использовать только там ,где без него не
работает, а также в инициализаторах.
2)Нашла в документации , что synchronize is deprecated  , и можно
больше не использовать . Говорят, что он автоматически синхронизирует,
только в сложных случаях лучше использовать )🤠

ZalyalovIldar commented:
Deprecated? O rly? 
https://developer.apple.com/reference/foundation/userdefaults

Да, synchronise используется автоматически, но причина, почему я прошу его проставить, потому что бывают случаи, что user default запоминает только в кэш и не успевает сделать синхронайз. 
Программируя, нужно стараться решать проблему для N-мерной задачи, чтобы приучить себя к порядку и во избежания возможных ошибок, а не потому что мне нравится, чтобы вы ставили synchronize() . Соотвественно я знаю, что бывают такие ошибки, и если они бывают, значит надо максимально обезопасить себя, а когда для этого нужно сделать всего лишь synchronize(), это очень круто и не так то и сложно) 
Если в следующей версии synchronize() станет депрекейтет официально, и прям вот эпл скажет, все чуваки, мы крутые, сделали очень удобно и у нас нет проблем с этим - ок, но пока давай лучше использовать это) 

Пруф: 

<img width="837" alt="2017-02-20 22 55 58" src="https://cloud.githubusercontent.com/assets/8490823/23140021/65f56396-f7c0-11e6-874d-a735b00c3ee3.png">

nataliSM committed:
fixed👩‍🔧
